cpp/tests/lists/combine/concatenate_rows_tests.cpp

```diff
@@ -486,3 +479,294 @@ TEST_F(ListConcatenateRowsTest, StringsColumnsWithEmptyListTest)
+                                          "",
+                                          "hhh",
+                                          "warp",
+                                          "donuts",
+                                          "parking",
+                                          "",
+                                          "apply",
+                                          "twelve",
+                                          "mouse",
+                                          "bbb",
+                                          "pom"};
+  std::vector<std::unique_ptr<cudf::column>> s1_children;
+  s1_children.push_back(s1_0.release());
+  s1_children.push_back(s1_1.release());
+  cudf::test::structs_column_wrapper s1(std::move(s1_children));
+  cudf::test::fixed_width_column_wrapper<int> l1_offsets{0, 0, 4, 7, 15, 15};
+  auto const l1_size = static_cast<cudf::column_view>(l1_offsets).size() - 1;
+  std::vector<bool> l1_validity{false, true, true, true, true};
+  auto l1 = cudf::make_lists_column(
+    l1_size,
+    l1_offsets.release(),
+    s1.release(),
+    1,
+    cudf::test::detail::make_null_mask(l1_validity.begin(), l1_validity.end()));
+
+  // concatenate_policy::IGNORE_NULLS
+  {
+    // perform the concatenate
+    cudf::table_view t({*l0, *l1});
+    auto result = cudf::lists::concatenate_rows(t, cudf::lists::concatenate_null_policy::IGNORE);
+
+    // expected
+    cudf::test::fixed_width_column_wrapper<int> se_0{
+      {10, 11, 12, 13, 2, 3, 4, 14, 15, 16, 5, 17, 18, 19, 20, 21, 22, 23, 24, 6, 7},
+      nulls_at({18})};
+    cudf::test::strings_column_wrapper se_1{
+      {"arg",    "mno",     "ampere", "gpu",   "bananas", "",      "",    "",    "hhh", "warp", "g",
+       "donuts", "parking", "",       "apply", "twelve",  "mouse", "bbb", "pom", "xyw", "ijk"},
+      nulls_at({5, 6})};
+    std::vector<std::unique_ptr<cudf::column>> se_children;
+    se_children.push_back(se_0.release());
+    se_children.push_back(se_1.release());
+    cudf::test::structs_column_wrapper se(std::move(se_children));
+    cudf::test::fixed_width_column_wrapper<int> le_offsets{0, 0, 4, 10, 19, 21};
+    auto const le_size = static_cast<cudf::column_view>(le_offsets).size() - 1;
+    std::vector<bool> le_validity{false, true, true, true, true};
+    auto expected = cudf::make_lists_column(
+      le_size,
+      le_offsets.release(),
+      se.release(),
+      1,
+      cudf::test::detail::make_null_mask(le_validity.begin(), le_validity.end()));
+
+    CUDF_TEST_EXPECT_COLUMNS_EQUAL(*result, *expected);
+  }
+
+  // concatenate_policy::NULLIFY_OUTPUT_ROW
+  {
+    // perform the concatenate
+    cudf::table_view t({*l0, *l1});
+    auto result =
+      cudf::lists::concatenate_rows(t, cudf::lists::concatenate_null_policy::NULLIFY_OUTPUT_ROW);
+
+    // expected
+    cudf::test::fixed_width_column_wrapper<int> se_0{{10, 11, 12, 13, 2, 3, 4, 14, 15, 16, 6, 7},
+                                                     nulls_at({})};
```

Comment:
It can, but I'd have to change the test to _EQUIVALENT instead of _EQUAL.  I'd prefer the stronger check to make sure the lists are being sanitized properly.